imageloading-core/src/main/java/com/google/accompanist/imageloading/ImageState.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.imageloading
+
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
+import androidx.compose.ui.unit.IntSize
+import kotlinx.coroutines.flow.Flow
+import kotlin.coroutines.cancellation.CancellationException
+
+/**
+ * A state base class that can be hoisted to control image loads for [rememberLoadPainter].
+ *
+ * @param shouldRefetchOnSizeChange Initial value for [shouldRefetchOnSizeChange].
+ */
+@Stable
+abstract class ImageState<R : Any>(
+    shouldRefetchOnSizeChange: (currentState: ImageLoadState, size: IntSize) -> Boolean,
```

Comment:
I am curious if we really still need ImageState public api at all? it was reasonable as a state for composable but now when we have a Painter maybe we can merge them, Painter is already a stateful thing. I would try to move shouldRefetchOnSizeChange param to rememberLoadPainter(). 
if we want to keep public val loadState we can make LoadPainter class public, introduce it there and return it from rememberLoadPainter().
wdyt?

Reply:
Yeah, PTAL. I'll probably finesse the API names, but the general structure feels good to me.